src/ImageSharp.Web/Middleware/ImageSharpMiddleware.cs

```diff
@@ -188,10 +188,10 @@ public async Task Invoke(HttpContext context)
             IDictionary<string, string> commands = this.requestParser.ParseRequestCommands(context);
             if (commands.Count > 0)
             {
-                // Strip out any unknown commands.
+                // Strip out any unknown commands
                 foreach (string command in new List<string>(commands.Keys))
                 {
-                    if (!this.knownCommands.Contains(command))
+                    if (!this.knownCommands.Contains(command, StringComparer.OrdinalIgnoreCase))
```

Comment:
I actually want command case to be respected. Reduces attack surface

Reply:
The ASP.NET Core runtime uses case insensitive query string parsing though (check the result of `Microsoft.AspNetCore.WebUtilities.QueryHelpers.ParseQuery("?test=true").Comparer` or https://github.com/dotnet/aspnetcore/issues/3450) and all the `IRequestParser` implementations in ImageSharp also return a case insensitive dictionary...

Besides that, the cache hash is already generated based on a [lower cased URI](https://github.com/SixLabors/ImageSharp.Web/blob/fa183578c358c63f3225dd76b3d0a92413f038f8/src/ImageSharp.Web/Middleware/ImageSharpMiddleware.cs#L538), so I don't see how this would reduce the attack surface.